package/src/external/reanimated/renderHelpers.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable reanimated/js-function-in-worklet */
+import type { Container } from "../../renderer/Container";
+import type { AnimatedProp, AnimatedProps } from "../../renderer/processors";
+import type { Node } from "../../dom/types";
+
+import {
+  startMapper,
+  stopMapper,
+  isSharedValue,
+  HAS_REANIMATED,
+} from "./moduleWrapper";
+
+const _bindings = new WeakMap<Node<unknown>, unknown>();
+
+export function sanitizeReanimatedProps(props: AnimatedProps<any>) {
+  if (!HAS_REANIMATED) {
+    return [props, {}];
+  }
+  const reanimatedProps = {} as AnimatedProps<any>;
+  const otherProps = {} as AnimatedProps<any>;
+  for (const propName in props) {
+    const propValue = props[propName];
+    if (isSharedValue(propValue)) {
```

Comment:
In reanimated, `isSharedValue` is implemented as
`
export function isSharedValue<T>(value: any): value is SharedValue<T> {
  'worklet';
  return typeof value === 'object' && value._isReanimatedSharedValue === true;
}`

However, `typeof null` is `object` 😅 so mentioning it in case you want to change the `isSharedValue` implementation upstream.

Reply:
Just noting this was fixed upstream here:
https://github.com/software-mansion/react-native-reanimated/pull/4128/